### ISSUE TITLE
GraphQL enum type with values that are reserved Swift keywords

### ISSUE THREAD

**Issue by attheodo** [apollographql/apollo-ios#53]
That's an interesting case. Our GraphQL backend has an `AlbumPrivacies` enum with values `PRIVATE` and `PUBLIC`. 

Reasonably, Apollo tries to generate the respective Swift enum in the following way:

```
/// The privacy modes for an album.
public enum AlbumPrivacies: String {
  case public = "PUBLIC" /// Public albums can be joined and viewed by anyone.
  case private = "PRIVATE" /// Private albums can be joined and viewed only with an invitation.
}
```

Unfortunately, `public` and `private` are of course reserved Swift keywords so unfortunately the generated code is broken.

I am not sure what's the most appropriate way to mitigate this on your side. Maybe check whether an enum value is a reserved keyword and prefix/suffix something to its name declaration? So in that case it could be `publicCase` and `privateCase` or something. 

Just thinking out loud here.

**Comment by MrAlek** [comment]
There's a closed issue on this here: https://github.com/apollographql/apollo-codegen/issues/33

**Comment by attheodo** [comment]
@MrAlek I guess it's missed. I am on `apollo-codegen 0.10.4` and doesn't seem to work.

**Comment by martijnwalraven** [comment]
Ah, that should be pretty easy to fix actually! Swift allows you to escape identifiers by using backticks. So in this case, you'd want the output to be:

```swift
/// The privacy modes for an album.
public enum AlbumPrivacies: String {
  case `public` = "PUBLIC" /// Public albums can be joined and viewed by anyone.
  case `private` = "PRIVATE" /// Private albums can be joined and viewed only with an invitation.
}
```

Unfortunately, it seems apollographql/apollo-codegen#33 doesn't take enum cases into account. But that should be a single line fix because the `escapeIdentifierIfNeeded` function has already been written.

**Comment by attheodo** [comment]
@martijnwalraven any estimates on when the fix could make it to `apollo-codegen` ?

**Comment by martijnwalraven** [comment]
@attheodo: Working on it right now, give me 15 min :)

**Comment by attheodo** [comment]
@martijnwalraven you Sir are awesome! Thanks!

**Comment by martijnwalraven** [comment]
@attheodo: Should be fixed in `apollo-codegen` 0.10.6.

**Comment by howlingpan** [comment]
It seems there are some special cases that are missing
```
query GetTimeline($count: Int!, $cursor: ID) {
    getTimeline(count: $count, cursor: $cursor) {
        list {
            ... on Poi {
                where
            }
        }
    }
}
```
will generate 
```
public var where: Coordinates {
            get {
              return snapshot["where"]! as! Coordinates
            }
            set {
              snapshot.updateValue(newValue, forKey: "where")
            }
          }
```
which raises a compiler error
I'm on ```apollo-codegen``` 0.17.0-alpha.14
